FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenDraftGmailAPIFlowTest.kt

```diff
@@ -0,0 +1,210 @@
+              draft.message.raw,
+              Base64.URL_SAFE or Base64.NO_PADDING or Base64.NO_WRAP
+            )
+            val mimeMessage = MimeMessage(
+              Session.getInstance(Properties()),
+              rawMimeMessageAsByteArray.inputStream()
+            )
+
+            return if (MESSAGE_SUBJECT == mimeMessage.subject) {
+              MockResponse()
+                .setResponseCode(HttpURLConnection.HTTP_OK)
+                .setBody(
+                  Draft().apply {
+                    factory = GsonFactory.getDefaultInstance()
+                    id = DRAFT_ID
+                    message = Message().apply {
+                      id = MESSAGE_ID
+                      threadId = THREAD_ID
+                      labelIds = listOf(LABEL_DRAFT)
+                    }
+                  }.toString()
+                )
+            } else {
+              MockResponse()
+                .setResponseCode(HttpURLConnection.HTTP_BAD_REQUEST)
+            }
+          }
+
+          request.path == "/gmail/v1/users/me/messages/$MESSAGE_ID?fields=id,threadId,historyId&format=full" -> {
+            return MockResponse()
+              .setResponseCode(HttpURLConnection.HTTP_OK)
+              .setBody(Message().apply {
+                factory = GsonFactory.getDefaultInstance()
+                id = MESSAGE_ID
+                threadId = THREAD_ID
+                labelIds = listOf(LABEL_DRAFT)
+                historyId = BigInteger.valueOf(Random.nextLong())
+              }.toString()).apply {
+                isDraftSaved = true
+              }
+          }
+
+          else -> return MockResponse().setResponseCode(HttpURLConnection.HTTP_NOT_FOUND)
+        }
+      }
+    })
+
+  private var isDraftSaved = false
+
+  @get:Rule
+  var ruleChain: TestRule = RuleChain
+    .outerRule(RetryRule.DEFAULT)
+    .around(ClearAppSettingsRule())
+    .around(GrantPermissionRuleChooser.grant(android.Manifest.permission.POST_NOTIFICATIONS))
+    .around(mockWebServerRule)
+    .around(addAccountToDatabaseRule)
+    .around(addPrivateKeyToDatabaseRule)
+    .around(
+      AddLabelsToDatabaseRule(
+        account = addAccountToDatabaseRule.account,
+        folders = listOf(
+          LocalFolder(
+            account = addAccountToDatabaseRule.account.email,
+            fullName = "Draft",
+            folderAlias = "Draft",
+            attributes = listOf("\\HasNoChildren", "\\Draft")
+          )
+        )
+      )
+    )
+    .around(activeActivityRule)
+    .around(ScreenshotTestRule())
+
+  @Test
+  fun testSavingDraftViaGmailAPI() {
+    activeActivityRule?.launch(intent)
+    registerAllIdlingResources()
+
+    onView(withId(R.id.editTextEmailSubject))
+      .check(matches(isDisplayed()))
+      .perform(
+        scrollTo(),
+        click(),
+        typeText(MESSAGE_SUBJECT)
+      )
+
+    Thread.sleep(DraftViewModel.DELAY_TIMEOUT * 2)
+    assertTrue(isDraftSaved)
```

Comment:
If I understand correctly, this only tests `isDraftSaved` flag gets true when user edits subject.
I think we need to add several other tests like mentioned in https://github.com/FlowCrypt/flowcrypt-ios/pull/1851?

Reply:
It tests the following scenario:

1. Open the compose screen
2. Type some text in the subject field
3. After that the app should create a new draft on a server
4. `assertTrue(isDraftSaved)` means that the requested `Gmail API` was called with a successful result and we saved a new draft.

> I think we need to add several other tests like mentioned in https://github.com/FlowCrypt/flowcrypt-ios/pull/1851?

Sorry, it's a little hard to understand `iOs` code for me. 

I just see the following comment

```
By the way, I think we should have UI test(for last case at above comment) which tests draft send feature

Create draft
Go back and open that draft
Update draft(update subject, message, recipient)
Send message
check if sent message is correct
```

Did you mean this one? 

@ioanmo226 Please add examples of other scenarios if you think they should be tested too.

Reply:
Please let me know if you have questions.

Reply:
Thinking... The above scenario looks too complex. It tests too many things that should be split. Also, the current Android realization and Espresso testing will not cover it without `dirty` code. On Android creating/updating a draft is a background task(out of the screen/Activity).

We can split the above scenario into the following
### Check the correct creating and updating
```
1. Create draft
2. Go back and create another draft
3. Check if 2 drafts are saved correctly
4. Update 1st draft. Go back to draft list screen and check if 1st draft is updated correctly and 2nd draft remains same
```

### Check the correct draft deleting
```
1. We have two drafts
2. Delete 1st draft, reload draft screen and check if 1st draft is deleted correctly. 
```

```
1. We have one drafts
2. Delete a draft and reload draft screen to check if draft folder is empty.
```

### Check the correct draft sending
```
1. Create new draft(only input subject) and go back and open that draft again
2. Update subject, insert message, recipients and click send button
3. Check if draft is sent correctly (Check if draft screen is empty after email send, go to sent folder and check if sent email is there)
```

Reply:
I agree

Reply:
I propose to merge the current PR that added a test for creating a draft and add new tests via a separate PR.
@ioanmo226 What do you think?

Reply:
I agree